SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc

```diff
@@ -0,0 +1,452 @@
+      continue;
+    }
+
+    float nelectrons = super_charge.energy();
+    // XXX -- Diffusion: using the center frame
+    if (diffusion_activated) {
+      auto migrated_charges = diffusion(position_at_pc, super_charge.energy(), drift_direction, half_pitch, thickness);
+      // remove the migrated charges
+      for (auto& mc : migrated_charges) {
+        nelectrons -= mc.energy();
+        // and convert back to the pixel ref. system
+        // Low-left origin/pitch -> relative within the pixel (a)
+        // Adding the pixel
+        const float pixel_x = current_pixel_int.first + (mc.x() + center_proxy_cell.x()) / pitch.first;
+        const float pixel_y = current_pixel_int.second + (mc.y() + center_proxy_cell.y()) / pitch.second;
+        const auto lp = pixdet->specificTopology().localPosition(MeasurementPoint(pixel_x, pixel_y));
+        mc.migrate_position(LocalPoint(lp.x(), lp.y(), mc.z()));
+      }
+      if (migrated_charges.size() > 0) {
+        LogDebug("Pixel3DDigitizerAlgorithm::drift") << "****************"
+                                                     << "MIGRATING (super-)charges"
+                                                     << "****************";
+        // Drift this charges on the other pixel
+        auto mig_colpoints = drift(hit, pixdet, bfield, migrated_charges, false);
+        LogDebug("Pixel3DDigitizerAlgorithm::drift") << "*****************"
+                                                     << "DOME MIGRATION"
+                                                     << "****************";
+      }
+    }
+
+    // Perform the drift, and check a potential lost of carriers because
+    // they reach the pasivation region (-z < thickness/2)
+    // XXX: not doing nothing, the carriers reach the electrode surface,
+    const float drift_distance = position_at_pc.perp() - _np_column_radius;
+
+    // Insert a charge loss due to Rad Damage here
+    // XXX: ??
+    float energyOnCollector = nelectrons;
+    // FIXME: is this correct?  Not for 3D...
+
+    if (pseudoRadDamage_ >= 0.001) {
+      const float module_radius = pixdet->surface().position().perp();
+      if (module_radius <= pseudoRadDamageRadius_) {
+        const float kValue = pseudoRadDamage_ / (module_radius * module_radius);
+        energyOnCollector = energyOnCollector * std::exp(-1.0 * kValue * drift_distance / max_radial_distance);
+      }
+    }
+    LogDebug("Pixel3DDigitizerAlgorithm::drift")
+        << "Drift distance = " << drift_distance / unit_um << " [um], "
+        << "Initial electrons = " << super_charge.energy()
+        << " [electrons], Electrons after loss/diff= " << energyOnCollector << " [electrons] ";
+    //DigitizerUtility::SignalPoint sp(CloudCenterX, CloudCenterY, Sigma_x, Sigma_y, hit.tof(), energyOnCollector);
+    // Load the Charge distribution parameters
+    // XXX -- probably makes no sense the SignalPoint anymore...
```

Comment:
then why do it?

Reply:
Please refer to [this comment](https://github.com/cms-sw/cmssw/pull/29018#discussion_r387023499), not sure yet, therefore keeping it until we verify it is actually not needed.